src/app/video-courses/video-course.service.ts

```diff
@@ -1,106 +1,101 @@
-      ' ultricies gravida.',
-      date: new Date((new Date()).getTime()),
-      duration: 70
-    },
-    {
-      id: 4,
-      title: 'Angular, React and Vue.js',
-      description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum mauris est,' +
-      ' convallis sed lobortis non, aliquam ac diam. Duis quis orci eros. Donec vitae nisl vulputate,' +
-      ' finibus risus ultricies, sollicitudin tortor. Suspendisse sollicitudin sed tellus id malesuada.' +
-      ' Praesent vitae laoreet justo, a scelerisque arcu. Fusce nec lorem mi. Donec non tortor erat.' +
-      ' Aliquam sit amet augue sed velit scelerisque sollicitudin. Proin gravida fermentum metus, nec laoreet' +
-      ' metus laoreet ut. Maecenas vitae dictum nulla. Nam interdum lorem quis ultrices rutrum. Nunc fermentum' +
-      ' sodales nisi, sed ultricies augue dignissim quis. Vestibulum id elit orci. Sed rhoncus mauris sed' +
-      ' ultricies gravida.',
-      date: new Date((new Date()).getTime() - 5000000000),
-      duration: 46
-    },
-    {
-      id: 5,
-      title: 'Mean stack',
-      description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum mauris est,' +
-      ' convallis sed lobortis non, aliquam ac diam. Duis quis orci eros. Donec vitae nisl vulputate,' +
-      ' finibus risus ultricies, sollicitudin tortor. Suspendisse sollicitudin sed tellus id malesuada.' +
-      ' Praesent vitae laoreet justo, a scelerisque arcu. Fusce nec lorem mi. Donec non tortor erat.' +
-      ' Aliquam sit amet augue sed velit scelerisque sollicitudin. Proin gravida fermentum metus, nec laoreet' +
-      ' metus laoreet ut. Maecenas vitae dictum nulla. Nam interdum lorem quis ultrices rutrum. Nunc fermentum' +
-      ' sodales nisi, sed ultricies augue dignissim quis. Vestibulum id elit orci. Sed rhoncus mauris sed' +
-      ' ultricies gravida.',
-      date: new Date((new Date()).getTime() - 1000000000),
-      duration: 30
-    },
-  ];
+  private readonly BASE_URL = 'http://localhost:3004/courses';
+  private courseList: VideoCourseItem[] = [];
+  private courseListChange = new Subject<VideoCourseItem[]>();
 
-  public getVideoCourses(): VideoCourseItem[] {
-    return this.courseList;
+  constructor(private httpClient: HttpClient) {
   }
 
-  public addCourse(course: VideoCourseItem): VideoCourseItem {
-    const list = this.courseList;
-    course.id = list.length ? list[list.length - 1].id + 1 : 1;
-    this.courseList = [...list, course];
+  public getCourseListChange(): Subject<VideoCourseItem[]> {
+    return this.courseListChange;
+  }
+
+  public getVideoCourses(): void {
+    if (this.courseList.length > 0) {
+      this.courseListChange.next(this.courseList);
+    } else {
+      this.fetchVideoCourses();
+    }
+  }
 
-    return course;
+  private fetchVideoCourses() {
+    this.httpClient.get<CourseItem[]>(`${this.BASE_URL}`)
+      .subscribe(response => {
+        this.courseList = response.map(course => {
```

Comment:
this.courseList = response.map(this.convertToVideoCourseItem);

Reply:
Done.